R/executionSettings.R

```diff
@@ -0,0 +1,167 @@
+
+# ExecutionSettings ----
+
+#' @description
+#' An R6 class to define an ExecutionSettings object
+#'
+#' @export
+ExecutionSettings <- R6::R6Class(
```

Comment:
@anthonysena I think this is a useful abstraction that can be used in almost all our packages and CohortGenerator seems an appropriate place to put it.

Some initial points that I think we need:

* Changing the generate cohorts api so that it creates this object OR uses it if specified - this way we can maintain compatibility for existing scripts. Alternatively we should provide dual APIs (old and new) by creating a new function `executeCohorts`.

* We will need to write a vignette

* we should update all functions in this package to use this

* We should look at the stratuegus execution context list to see if there is additional stuff we should replicate there

* Perhaps we also move or replicate the cdm source meta data function from strategus here?

* We should add a helper function to create an instance

Reply:
Agreed @azimov and thanks for this contribution @mdlavallee92!

> Changing the generate cohorts api so that it creates this object OR uses it if specified - this way we can maintain compatibility for existing scripts. Alternatively we should provide dual APIs (old and new) by creating a new function executeCohorts.

Let's aim to add this as an additional way of encapsulating the settings as you suggested and then fully adopt it when we have the opportunity to break compatibility with v1.x.